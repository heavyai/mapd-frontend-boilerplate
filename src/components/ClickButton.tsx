import React from 'react'

import { Dispatcher } from '~/actions'
import { addClick } from '~/actions/clicks'

const styles = require( './ClickButton.css' )


type Props = {
  dispatch: Dispatcher
}

const ClickButton: React.StatelessComponent<Props> = ( { dispatch } ) => {
  const click = () => {
    dispatch( addClick() )
  }

  return (
    <button className={ styles.button } onClick={ click }>
      The Button
    </button>
  )
}


export default ClickButton
