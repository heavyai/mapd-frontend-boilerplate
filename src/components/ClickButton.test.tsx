import React from 'react'
import * as renderer from 'react-test-renderer'

import ClickButton from './ClickButton'


test( 'Component renders', () => {
  const component = renderer.create(
    <ClickButton dispatch={ jest.fn() } />
  )

  expect( component.toJSON() ).toMatchSnapshot()
} )
