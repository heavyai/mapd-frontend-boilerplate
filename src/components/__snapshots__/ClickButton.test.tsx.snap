// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component renders 1`] = `
<button
  className="button"
  onClick={[Function]}
>
  The Button
</button>
`;
